File: rmi/java/src/org/Book.java
package org;

import java.rmi.server.UnicastRemoteObject;
import java.util.ArrayList;

public class Book extends UnicastRemoteObject implements BOOK {
    public String name;
    ArrayList<Page> p;
    public Book() throws Exception {
        this.name = "I am just a little book";
        p = new ArrayList<>();
    }
    public Book(String name) throws Exception {
        this.name = name;
        p = new ArrayList<>();
    }
    public void add(String content) throws Exception {
        Page p = new Page(content);
        this.p.add(p);
    }
    public void show() throws Exception {
        System.out.println(name);
        for (Page p: this.p) {
            System.out.println(p.read());
        }
    }
}

File: rmi/py/Book.py
from Page import Page
import Pyro4

@Pyro4.expose
class Book(object):
    def __init__(self, n = "I am a little book"):
        self.n = n
        self.p = []

    def add(self, c):
        p = Page(c)
        self.p += [p]

    def show(self):
        print(self.n)
        for p in self.p:
            print(p.read())
